lib/dateTime.js

```diff
@@ -0,0 +1,580 @@
+				['dom', 'seg', 'ter', 'qua', 'qui', 'sex', 'sáb'],
+				['D', 'S', 'T', 'Q', 'Q', 'S', 'S']
+			];
+			break;
+		case 'sv':
+			dateFormats = ['dddd \'den\' d MMMM yyyy', 'd MMMM yyyy', 'yyyy-MM-dd', 'MMMM yyyy', 'dd MMMM'];
+			dayPeriods = ['fm', 'em'];
+			months = [
+				['Januari', 'Februari', 'Mars', 'April', 'Maj', 'Juni', 'Juli', 'Augusti', 'September', 'Oktober', 'November', 'December'],
+				['Jan', 'Feb', 'Mar', 'Apr', 'Maj', 'Jun', 'Jul', 'Aug', 'Sep', 'Okt', 'Nov', 'Dec']
+			];
+			days = [
+				['Söndag', 'Måndag', 'Tisdag', 'Onsdag', 'Torsdag', 'Fredag', 'Lördag'],
+				['Sön', 'Mån', 'Tis', 'Ons', 'Tor', 'Fre', 'Lör'],
+				['S', 'M', 'T', 'O', 'T', 'F', 'L']
+			];
+			break;
+		case 'tr':
+			dateFormats = ['dd MMMM yyyy dddd', 'dd MMMM yyyy', 'dd.MM.yyyy', 'MMMM yyyy', 'dd MMMM'];
+			dayPeriods = ['ÖÖ', 'ÖS'];
+			months = [
+				['Ocak', 'Şubat', 'Mart', 'Nisan', 'Mayıs', 'Haziran', 'Temmuz', 'Ağustos', 'Eylül', 'Ekim', 'Kasım', 'Aralık'],
+				['Oca', 'Şub', 'Mar', 'Nis', 'May', 'Haz', 'Tem', 'Ağu', 'Eyl', 'Ek', 'Kas', 'Ara']
+			];
+			days = [
+				['Pazar', 'Pazartesi', 'Salı', 'Çarşamba', 'Perşembe', 'Cuma', 'Cumartesi'],
+				['Paz', 'Pzt', 'Sal', 'Çar', 'Per', 'Cum', 'Cmt'],
+				['P', 'P', 'S', 'Ç', 'P', 'C', 'C']
+			];
+			break;
+		case 'zh':
+			dateFormats = ['yyyy年M月d日', 'yyyy年M月d日', 'yyyy/M/d', 'yyyy年M月', 'M月d日'];
+			dayPeriods = ['上午', '下午'];
+			months[0] = months[1] = ['一月', '二月', '三月', '四月', '五月', '六月', '七月', '八月', '九月', '十月', '十一月', '十二月'];
+			days[0] = days[1] = ['週日', '週一', '週二', '週三', '週四', '週五', '週六'];
+			days[2] = ['日', '一', '二', '三', '四', '五', '六'];
+			break;
+	}
+
+	switch (language) {
+		case 'en-gb':
+			dateFormats = ['dddd, d MMMM yyyy', 'dd MMMM yyyy', 'dd/MM/yyyy', 'MMMM yyyy', 'd MMMM'];
+			break;
+		case 'fr-ca':
+			dateFormats[1] = 'MMM d yyyy';
+			dateFormats[2] = 'yyyy-MM-dd';
+			dateFormats[4] = 'MMMM d';
+			firstDayOfWeek = 0;
+			break;
+		case 'zh-cn':
+			fullTimeFormat = `ZZZ ${timeFormat}`;
+			break;
+	}
+
+	const descriptor = {
+		hour24: hour24,
+		formats: {
+			dateFormats: {
+				'full': dateFormats[0],
+				'medium': dateFormats[1],
+				'short': dateFormats[2],
+				'monthYear': dateFormats[3],
+				'monthDay': dateFormats[4]
+			},
+			timeFormats: {
+				'full': fullTimeFormat,
+				'medium': timeFormat,
+				'short': timeFormat
+			}
+		},
+		calendar: {
+			firstDayOfWeek: firstDayOfWeek,
+			weekendStartDay: weekendStartDay,
+			weekendEndDay: weekendEndDay,
+			months: {
+				short: months[1],
+				long: months[0]
+			},
+			days: {
+				narrow: days[2],
+				short: days[1],
+				long: days[0]
+			},
+			dayPeriods: {am: dayPeriods[0], pm: dayPeriods[1]}
+		}
+	};
+
+	const settings = getDocumentLocaleSettings();
+	if (settings.overrides.date) {
+		merge(descriptor, settings.overrides.date);
+	}
+
+	// TODO: fix 12/24 hour clock bug with missing info in format string
```

Comment:
Is this still applicable? Should there be an issue for it?

Reply:
Yeah good call. I'd originally thought I'd fix this existing bug as part of this work, but I think I'll do it later -- I'll create an issue.

Reply:
Created #32.